gateway/src/apicast/proxy.lua

```diff
@@ -300,10 +302,19 @@ function _M:rewrite(service, context)
   context.credentials = ctx.credentials
 end
 
-function _M:access(service, usage, credentials, ttl)
+function _M:access(service)
   local ctx = ngx.ctx
+  local usage = ctx.usage
+
+  -- If routing policy changes the upstream and it only belongs to a specified
+  -- owner, we need to filter out the usage for APIs that are not used at all.
+  if ctx.context.route_upstream then
```

Comment:
I think that we need to do this in the batching policy as well.

The batching policy acts in the access phase and grabs the usage in the context. When the batching policy is configured, we make sure that the apicast policy does not act in the access phase too by setting a flag. This is to avoid authorizing in both of them https://github.com/3scale/APIcast/blob/f02d7d89c521db910c7bf885b13097cbf227a3ae/gateway/src/apicast/policy/3scale_batcher/3scale_batcher.lua#L187